src/hash/clu_alg_hash.c

```diff
@@ -24,20 +24,39 @@
 
 #define MAX_BUFSIZE 8192
 
-int wolfCLU_md5Setup(int argc, char** argv)
+int wolfCLU_algHashSetup(int argc, char** argv)
 {
     WOLFSSL_BIO *bioIn  = NULL;
     WOLFSSL_BIO *bioOut = NULL;
     int     ret         = 0;
 
-#ifdef NO_MD5
-    WOLFCLU_LOG(WOLFCLU_E0, "wolfCrypt compiled without MD5 support");
-    ret = NOT_COMPILED_IN;
-    (void)bioIn;
-    (void)bioOut;
-    (void) argc; /* silence unused variable warning */
-    (void) argv; /* silence unused variable warning */
-#else
+const char* algs[]  =   {   /* list of acceptable algorithms */
+#ifndef NO_MD5
+        "md5",
+#endif
+#ifndef NO_SHA256
+        "sha256",
+#endif
+#ifdef WOLFSSL_SHA384
+        "sha384",
+#endif
+#ifdef WOLFSSL_SHA512
+        "sha512",
+#endif
+        NULL /* terminal element (also stops the array from being 0-size */
+    };
+    size_t algsSz = sizeof(algs) / sizeof(algs[0]) - 1; /* -1 to ignore NULL */
+
+    char*   alg;                /* algorithm being used */
+    int     size    =   0;      /* message digest size */
+    int     i          =   0;   /* loop variable */
```

Comment:
check the spacing on this / maybe hard tabs?

Reply:
Done